### ISSUE TITLE
[BUG] Windows build error: _librsyncmodule.obj : error LNK2001: unresolved external symbol rs_...

### ISSUE THREAD

**Issue by t9t** [rdiff-backup/rdiff-backup#443]
## Bug summary

When building on Windows following [these instructions](https://github.com/rdiff-backup/rdiff-backup/blob/master/docs/Windows-DEVELOP.md), I get the error `_librsyncmodule.obj : error LNK2001: unresolved external symbol rs_...` (for multiple symbols).

## Version, Python, Operating System

rdiff-backup `master` as of right now (`f858f01c917854e3bce1741297c50a69e4d958e0`), Python 3.7.8, Windows 10 (2004).

## More information

As promised, I want to try and see if I can help out with getting some of the test working on/for Windows (#347). So to start I'm trying to build the app op Windows.

I'm trying to get it working on my Windows machine using [the instructions on this page](https://github.com/rdiff-backup/rdiff-backup/blob/master/docs/Windows-DEVELOP.md) and after having installed all the dependencies and building librsync, I'm unfortunately stuck on a build issue when running `python setup.py build`. I have some Python experience (mainly webapps), but no experience integrating with C libraries, so I'm kind of stuck here unsure how to continue.

I have Python and the VC++ build tools installed and built librsync (version 2.2.1 as indicated on the aforementioned page). The files in `c:\librsync` are:
```
include/librsync.h
include/librsync_export.h
lib/rsync.lib
share/man/man1/rdiff.1
share/man/man3/librsync.3
```

I would expect a DLL file there somewhere, but then again I'm not sure.

When I try to build rdiff-backup, I get the following error:

```
> python setup.py build
running build
running build_py
running build_templates
running build_ext
building 'rdiff_backup._librsync' extension
C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.26.28801\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Drsync_EXPORTS -IC:\librsync\include "-IC:\Program Files\Python37\include" "-IC:\Program Files\Python37\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.26.28801\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\winrt" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\cppwinrt" /Tcsrc/_librsyncmodule.c /Fobuild\temp.win-amd64-3.7\Release\src/_librsyncmodule.obj
_librsyncmodule.c
C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.26.28801\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:C:\librsync\lib "/LIBPATH:C:\Program Files\Python37\libs" "/LIBPATH:C:\Program Files\Python37\PCbuild\amd64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.26.28801\lib\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.18362.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.18362.0\um\x64" rsync.lib /EXPORT:PyInit__librsync build\temp.win-amd64-3.7\Release\src/_librsyncmodule.obj /OUT:build\lib.win-amd64-3.7\rdiff_backup\_librsync.cp37-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.7\Release\src\_librsync.cp37-win_amd64.lib
   Creating library build\temp.win-amd64-3.7\Release\src\_librsync.cp37-win_amd64.lib and object build\temp.win-amd64-3.7\Release\src\_librsync.cp37-win_amd64.exp
_librsyncmodule.obj : error LNK2001: unresolved external symbol rs_sig_begin
_librsyncmodule.obj : error LNK2001: unresolved external symbol rs_job_free
_librsyncmodule.obj : error LNK2001: unresolved external symbol rs_loadsig_begin
_librsyncmodule.obj : error LNK2001: unresolved external symbol rs_file_copy_cb
_librsyncmodule.obj : error LNK2001: unresolved external symbol rs_job_iter
_librsyncmodule.obj : error LNK2001: unresolved external symbol rs_build_hash_table
_librsyncmodule.obj : error LNK2001: unresolved external symbol rs_delta_begin
_librsyncmodule.obj : error LNK2001: unresolved external symbol rs_free_sumset
_librsyncmodule.obj : error LNK2001: unresolved external symbol rs_patch_begin
build\lib.win-amd64-3.7\rdiff_backup\_librsync.cp37-win_amd64.pyd : fatal error LNK1120: 9 unresolved externals
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.26.28801\\bin\\HostX86\\x64\\link.exe' failed with exit status 1120

```

**Comment by t9t** [comment]
I peeked into [tools/windows/tasks/get-librsync-tarball.yml](https://github.com/rdiff-backup/rdiff-backup/blob/master/tools/windows/tasks/get-librsync-tarball.yml) and see there is a different cmake configuration command line than on the instructions page. When I use that (but omit `-G "Visual Studio 15 2017"`, because I installed the "16 2019" version) it builds librsync in a way that actually makes the rdiff-backup build succeed.

```
cmake -A x64 -D BUILD_RDIFF=OFF . -D CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=TRUE -D BUILD_SHARED_LIBS=TRUE -DCMAKE_INSTALL_PREFIX=c:\librsync
```

So probably the instructions page can just be updated with this command like.
